### PR TITLE
Add first user endpoint

### PR DESCRIPTION
**Pull request by amulware**
Please see what you think of this approach using structs to get endpoints like:

```csharp
endpoints.User.Get
```

Other endpoints will look something like:

```csharp
endpoints.Projects.GetAll
endpoints.Projects.Create(project) or endpoints.Projects.Create(workspaceId)
endpoints.Projects.Update(project) or endpoints.Projects.Create(workspaceId, projectId)
```

The same interface could be done using reference types as well, though I would then initialize the endpoint-tree once, and not create them when needed like here.

### CODE REVIEW

File: Toggl.Ultrawave/Endpoints/Endpoint.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        private Endpoint(Uri baseUrl, string uri, HttpMethod method)
+            : this(new Uri(baseUrl, uri), method)
+        {
+        }
+
+        private Endpoint(Uri url, HttpMethod method)
+        {
+            Ensure.ArgumentIsNotNull(url, nameof(url));
+            if (!url.IsAbsoluteUri)
```
Comment (willsb) [inline Toggl.Ultrawave/Endpoints/Endpoint.cs]: I'd refactor this check to a method inside Ensure.

File: Toggl.Ultrawave/Endpoints/EndpointCollection.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using Toggl.Ultrawave.Helpers;
+
+namespace Toggl.Ultrawave.Endpoints
+{
+    internal sealed class EndpointCollection
```
Comment (willsb) [inline Toggl.Ultrawave/Endpoints/EndpointCollection.cs]: I don't like the name Collection here. Just `Endpoints` or `ApiEndpoints` would be better.

File: Toggl.Ultrawave/Endpoints/Endpoint.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        public static Endpoint Get(Uri baseUrl, string relativeUri)
+            => new Endpoint(baseUrl, relativeUri, HttpMethod.Get);
+
+        public static Endpoint Put(Uri baseUrl, string relativeUri)
+            => new Endpoint(baseUrl, relativeUri, HttpMethod.Put);
+
+        public static Endpoint Post(Uri baseUrl, string relativeUri)
+            => new Endpoint(baseUrl, relativeUri, HttpMethod.Post);
```
Comment (willsb) [inline Toggl.Ultrawave/Endpoints/Endpoint.cs]: Add stuff for `Patch` and `Delete` as well.

File: Toggl.Ultrawave/Endpoints/Endpoint.cs (hunk 1 above)
Comment (amulware) [inline Toggl.Ultrawave/Endpoints/Endpoint.cs]: Like, `Ensure.IsTrue()`?

File: Toggl.Ultrawave/Endpoints/Endpoint.cs (hunk 3 above)
Comment (amulware) [inline Toggl.Ultrawave/Endpoints/Endpoint.cs]: Good point.

File: Toggl.Ultrawave/Endpoints/EndpointCollection.cs (hunk 2 above)
Comment (amulware) [inline Toggl.Ultrawave/Endpoints/EndpointCollection.cs]: I don't really like `ApiEndpoints`, but the namespace is already called `Endpoints`. :/

File: Toggl.Ultrawave/Endpoints/Endpoint.cs (hunk 1 above)
Comment (willsb) [inline Toggl.Ultrawave/Endpoints/Endpoint.cs]: `Ensure.UriIsAbsolute` is fine. I don't see why checks can't be specific.

**Comment by amulware** [comment]
All comments addressed as discussed. Squash at will after due process.

File: Toggl.Multivac/Ensure.cs [hunk 4]
```diff
@@ -13,5 +13,12 @@ public static void ArgumentIsNotNull<T>(T value, string argumentName)
             throw new ArgumentNullException(argumentName);
         }
 
+        public static void UriIsAbsolute(Uri uri, string argumentName)
+        {
+            if (uri.IsAbsoluteUri)
```
Comment (willsb) [inline Toggl.Multivac/Ensure.cs]: We need to set guidelines for early return statements and then enforce it. This is fine for the time being, though.

**Pull request merged**